ui/src/components/FormModifications/FormModifications.test.tsx

```diff
@@ -0,0 +1,213 @@
+        firstStandardTextField.field
+    );
+    const [component2ParentElement, component2Input] = getAndVerifyHTMLTextComponentsForField(
+        secondStandardTextField.field
+    );
+
+    const componentMakingModsTextBox1 = getAndVerifyHTMLTextComponentsForField(
+        firstModificationField.field
+    )[1];
+
+    const mods1Field1 = findMods(
+        firstModificationField,
+        firstValueToInput,
+        firstStandardTextField.field
+    );
+    const mods1Field2 = findMods(
+        firstModificationField,
+        firstValueToInput,
+        secondStandardTextField.field
+    );
+
+    const mods2Field1 = findMods(
+        firstModificationField,
+        secondValueToInput,
+        firstStandardTextField.field
+    );
+    const mods2Field2 = findMods(
+        firstModificationField,
+        secondValueToInput,
+        secondStandardTextField.field
+    );
+
+    await userEvent.type(componentMakingModsTextBox1, firstValueToInput);
+
+    const verifyAllProps = (
+        parentElement: Element,
+        input: Element,
+        mods?: { value?: string | number | boolean; help?: string; label?: string }
+    ) => {
+        expect(input).toHaveAttribute('value', mods?.value);
+
+        invariant(typeof mods?.help === 'string');
+        expect(parentElement.textContent?.includes(mods.help)).toEqual(true);
+
+        invariant(typeof mods?.label === 'string');
+        expect(parentElement.textContent?.includes(mods.label)).toEqual(true);
+    };
+
+    expect(componentInput).toHaveAttribute('disabled');
+    verifyAllProps(componentParentElement, componentInput, mods1Field1);
+
+    expect(component2Input).toHaveAttribute('disabled');
+    verifyAllProps(component2ParentElement, component2Input, mods1Field2);
+
+    await userEvent.type(componentMakingModsTextBox1, secondValueToInput);
+
+    expect(componentInput).not.toHaveAttribute('disabled');
+    verifyAllProps(componentParentElement, componentInput, mods2Field1);
+
+    expect(component2Input).not.toHaveAttribute('disabled');
+    verifyAllProps(component2ParentElement, component2Input, mods2Field2);
+});
+
+it('verify markdown modifications', async () => {
+    setUpConfigWithDefaultValue();
+    renderModalWithProps(props);
+
+    const firstValueToInput = 'a';
+    const secondValueToInput = 'aa';
+
+    const [componentParentElement] = getAndVerifyHTMLTextComponentsForField(
+        firstStandardTextField.field
+    );
+    const [component2ParentElement] = getAndVerifyHTMLTextComponentsForField(
+        secondStandardTextField.field
+    );
+
+    const componentMakingModsTextBox1 = getAndVerifyHTMLTextComponentsForField(
+        secondModificationField.field
+    )[1];
+
+    const firstElementOuterHTMLBeforeMods = componentParentElement?.outerHTML;
+
+    const mods1Field2 = findMods(
+        secondModificationField,
+        firstValueToInput,
+        secondStandardTextField.field
+    );
+
```

Comment:
```suggestion
    invariant(mods1Field2.markdownMessage?.markdownType === 'hybrid');
    invariant(mods1Field2.markdownMessage.text);
```

Reply:
Colocate all assumptions regarding the field here.

Reply:
done